sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -76,62 +127,20 @@ public RequestFailedException(int status, string message, Exception? innerExcept
         [EditorBrowsable(EditorBrowsableState.Never)]
         public RequestFailedException(int status, string message, string? errorCode, Exception? innerException)
-            : base(message, innerException)
+            : base(message, default, innerException)
         {
             Status = status;
             ErrorCode = errorCode;
         }
 
-        private RequestFailedException(int status, (string Message, ResponseError? Error) details) :
-            this(status, details.Message, details.Error?.Code, null)
-        {
-        }
-
-        private RequestFailedException(int status, ErrorDetails details, Exception? innerException) :
-            this(status, details.Message, details.ErrorCode, innerException)
-        {
-            if (details.Data != null)
-            {
-                foreach (KeyValuePair<string, string> keyValuePair in details.Data)
-                {
-                    Data.Add(keyValuePair.Key, keyValuePair.Value);
-                }
-            }
-        }
+        #endregion
 
-        /// <summary>Initializes a new instance of the <see cref="RequestFailedException"></see> class
-        /// with an error message, HTTP status code, and error code obtained from the specified response.</summary>
-        /// <param name="response">The response to obtain error details from.</param>
-        public RequestFailedException(Response response)
-            : this(response, null)
-        {
-        }
-
-        /// <summary>Initializes a new instance of the <see cref="RequestFailedException"></see> class
-        /// with an error message, HTTP status code, and error code obtained from the specified response.</summary>
-        /// <param name="response">The response to obtain error details from.</param>
-        /// <param name="innerException">An inner exception to associate with the new <see cref="RequestFailedException"/>.</param>
-        public RequestFailedException(Response response, Exception? innerException)
-            : this(response, innerException, null)
-        {
-        }
-
-        /// <summary>Initializes a new instance of the <see cref="RequestFailedException"></see> class
-        /// with an error message, HTTP status code, and error code obtained from the specified response.</summary>
-        /// <param name="response">The response to obtain error details from.</param>
-        /// <param name="innerException">An inner exception to associate with the new <see cref="RequestFailedException"/>.</param>
-        /// <param name="detailsParser">The parser to use to parse the response content.</param>
-        public RequestFailedException(Response response, Exception? innerException, RequestFailedDetailsParser? detailsParser)
-            : this(response.Status, CreateExceptionDetails(response, detailsParser), innerException)
-        {
-            _response = response;
-        }
+        #region ISerializable implementation
```

Comment:
For example, @KrzysztofCwalina, we could remove the serialization constructor from the base exception type in ClientModel if we wanted to, and add it back if/when we did the Azure.Core 2.0 integration.